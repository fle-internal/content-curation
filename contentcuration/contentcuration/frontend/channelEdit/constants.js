import { fileErrors } from 'shared/constants';

export const RouterNames = {
  SANDBOX: 'SANDBOX',
  TREE_ROOT_VIEW: 'TREE_ROOT_VIEW',
  TREE_VIEW: 'TREE_VIEW',
  STAGING_TREE_VIEW: 'STAGING_TREE_VIEW',
  CONTENTNODE_DETAILS: 'CONTENTNODE_DETAILS',
  MULTI_CONTENTNODE_DETAILS: 'MULTI_CONTENTNODE_DETAILS',
  IMPORT_FROM_CHANNELS: 'IMPORT_FROM_CHANNELS',
  IMPORT_FROM_CHANNELS_BROWSE: 'IMPORT_FROM_CHANNELS_BROWSE',
  IMPORT_FROM_CHANNELS_SEARCH: 'IMPORT_FROM_CHANNELS_SEARCH',
  IMPORT_FROM_CHANNELS_REVIEW: 'IMPORT_FROM_CHANNELS_REVIEW',
  ADD_TOPICS: 'ADD_TOPICS',
  ADD_EXERCISE: 'ADD_EXERCISE',
  UPLOAD_FILES: 'UPLOAD_FILES',
  TRASH: 'TRASH',
  ADD_PREVIOUS_STEPS: 'ADD_PREVIOUS_STEPS',
  ADD_NEXT_STEPS: 'ADD_NEXT_STEPS',
};

export const viewModes = {
  DEFAULT: 'DEFAULT_VIEW',
  COMFORTABLE: 'COMFORTABLE_VIEW',
  COMPACT: 'COMPACT_VIEW',
};

export const ValidationErrors = {
  TITLE_REQUIRED: 'TITLE_REQUIRED',
  LICENCE_REQUIRED: 'LICENCE_REQUIRED',
  COPYRIGHT_HOLDER_REQUIRED: 'COPYRIGHT_HOLDER_REQUIRED',
  LICENCE_DESCRIPTION_REQUIRED: 'LICENCE_DESCRIPTION_REQUIRED',
  MASTERY_MODEL_REQUIRED: 'MASTERY_MODEL_REQUIRED',
  MASTERY_MODEL_INVALID: 'MASTERY_MODEL_INVALID',
  QUESTION_REQUIRED: 'QUESTION_REQUIRED',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
  NO_VALID_PRIMARY_FILES: 'NO_VALID_PRIMARY_FILES',
  ...fileErrors,
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemToolbarActions = {
  EDIT_ITEM: 'EDIT_ITEM',
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  DELETE_ITEM: 'DELETE_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'questionTypeSingleSelection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'questionTypeMultipleSelection',
  [AssessmentItemTypes.TRUE_FALSE]: 'questionTypeTrueFalse',
  [AssessmentItemTypes.INPUT_QUESTION]: 'questionTypeInput',
};

export const TabNames = {
  DETAILS: 'details',
  PREVIEW: 'preview',
  QUESTIONS: 'questions',
  RELATED: 'related',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};
