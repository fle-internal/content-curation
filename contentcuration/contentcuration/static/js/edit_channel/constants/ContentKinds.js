module.exports = [
  {
    kind: 'document',
  },
  {
    kind: 'topic',
  },
  {
    kind: 'video',
  },
  {
    kind: 'html5',
  },
  {
    kind: 'audio',
  },
  {
    kind: 'exercise',
  },
];
