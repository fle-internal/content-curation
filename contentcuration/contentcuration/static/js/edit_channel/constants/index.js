const ContentKinds = require('./ContentKinds');
const FormatPresets = require('./FormatPresets');
const Languages = require('./Languages');
const Licenses = require('./Licenses');
const MasteryModels = require('./MasteryModels');
const Roles = require('./Roles');

module.exports = {
  ContentKinds,
  FormatPresets,
  Languages,
  Licenses,
  MasteryModels,
  Roles,
};
