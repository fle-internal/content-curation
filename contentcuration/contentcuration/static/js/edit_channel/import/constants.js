export const PageTypes = {
  TREE_VIEW: 'TREE_VIEW',
  IMPORT_PREVIEW: 'IMPORT_PREVIEW',
  SEARCH_RESULTS: 'SEARCH_RESULTS',
};
