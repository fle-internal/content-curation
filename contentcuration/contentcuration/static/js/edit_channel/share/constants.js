export const Permissions = {
  OWNER: 'manage',
  EDIT: 'edit',
  VIEW_ONLY: 'view'
};

export const PermissionRanks = [
	{
		"shareMode": Permissions.VIEW_ONLY,
		"rank": 0,
		"field": 'viewers'
	},
	{
		"shareMode": Permissions.EDIT,
		"rank": 1,
		"field": 'editors'
	},
	{
		"shareMode": Permissions.OWNER,
		"rank": 2,
		"field": null
	},
]
