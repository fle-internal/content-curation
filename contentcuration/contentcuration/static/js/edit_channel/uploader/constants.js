export const TabNames = {
  DETAILS: 'DETAILS',
  PREVIEW: 'PREVIEW',
  QUESTIONS: 'QUESTIONS',
  PREREQUISITES: 'PREREQUISITES',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};

export const AssessmentItemToolbarActions = {
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  EDIT_ITEM: 'EDIT_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
  DELETE_ITEM: 'DELETE_ITEM',
};

export const AssessmentItemValidationErrors = {
  BLANK_QUESTION: 'BLANK_QUESTION',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'Single selection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'Multiple selection',
  [AssessmentItemTypes.TRUE_FALSE]: 'True/False',
  [AssessmentItemTypes.INPUT_QUESTION]: 'Input question',
};
