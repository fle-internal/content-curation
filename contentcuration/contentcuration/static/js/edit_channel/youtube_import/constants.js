export const PageTypes = {
  SUBMIT_URL: 'SUBMIT_URL',
  CONFIRM: 'CONFIRM',
};

export const ImportStatus = {
  IDLE: 'IDLE',
  EXTRACTING: 'EXTRACTING',
  EXTRACT_ERROR: 'EXTRACT_ERROR',
  DOWNLOADING: 'DOWNLOADING',
  DOWNLOAD_ERROR: 'DOWNLOAD_ERROR',
  START_IMPORT: 'START_IMPORT',
  DONE: 'DONE',
};
