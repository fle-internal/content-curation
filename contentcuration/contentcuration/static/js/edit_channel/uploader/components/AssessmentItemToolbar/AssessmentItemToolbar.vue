<template>
  <span>
    <VTooltip top>
      <template slot="activator" slot-scope="{ on }">
        <VBtn
          icon
          :disabled="!canMoveUp"
          data-test="toolbarIconArrowUp"
          v-on="on"
          @click="clickItem(actions.MOVE_ITEM_UP)"
        >
          <VIcon :color="arrowUpColor">
            keyboard_arrow_up
          </VIcon>
        </VBtn>
      </template>
      <span>Move up</span>
    </VTooltip>

    <VTooltip top>
      <template slot="activator" slot-scope="{ on }">
        <VBtn
          icon
          :disabled="!canMoveDown"
          data-test="toolbarIconArrowDown"
          v-on="on"
          @click="clickItem(actions.MOVE_ITEM_DOWN)"
        >
          <VIcon :color="arrowDownColor">
            keyboard_arrow_down
          </VIcon>
        </VBtn>
      </template>
      <span>Move down</span>
    </VTooltip>

    <VTooltip v-if="displayEdit" top>
      <template slot="activator" slot-scope="{ on }">
        <VBtn
          icon
          data-test="toolbarIconEdit"
          v-on="on"
          @click="clickItem(actions.EDIT_ITEM)"
        >
          <VIcon color="#686868">
            edit
          </VIcon>
        </VBtn>
      </template>
      <span>Edit</span>
    </VTooltip>

    <VMenu bottom left>
      <template slot="activator" slot-scope="{ on }">
        <VBtn icon v-on="on">
          <VIcon color="#686868">
            more_vert
          </VIcon>
        </VBtn>
      </template>

      <VList>
        <VListTile
          data-test="toolbarMenuAddItemAbove"
          @click="clickItem(actions.ADD_ITEM_ABOVE)"
        >
          <VListTileTitle>Add question above</VListTileTitle>
        </VListTile>

        <VListTile
          data-test="toolbarMenuAddItemBelow"
          @click="clickItem(actions.ADD_ITEM_BELOW)"
        >
          <VListTileTitle>Add question below</VListTileTitle>
        </VListTile>

        <VListTile
          data-test="toolbarMenuDeleteItem"
          @click="clickItem(actions.DELETE_ITEM)"
        >
          <VListTileTitle>Delete</VListTileTitle>
        </VListTile>
      </VList>
    </VMenu>
  </span>
</template>

<script>

  import { AssessmentItemToolbarActions } from '../../constants';

  const GRAY_DARK = '#686868';
  const GRAY_LIGHT = '#9c9c9c';

  export default {
    name: 'AssessmentItemToolbar',
    props: {
      canMoveUp: {
        type: Boolean,
        default: true,
      },
      canMoveDown: {
        type: Boolean,
        default: true,
      },
      displayEdit: {
        type: Boolean,
        default: true,
      },
    },
    data() {
      return {
        actions: AssessmentItemToolbarActions,
      };
    },
    computed: {
      arrowUpColor() {
        if (this.canMoveUp) {
          return GRAY_DARK;
        } else {
          return GRAY_LIGHT;
        }
      },
      arrowDownColor() {
        if (this.canMoveDown) {
          return GRAY_DARK;
        } else {
          return GRAY_LIGHT;
        }
      },
    },
    methods: {
      clickItem(action) {
        this.$emit('click', action);
      },
    },
  };

</script>
