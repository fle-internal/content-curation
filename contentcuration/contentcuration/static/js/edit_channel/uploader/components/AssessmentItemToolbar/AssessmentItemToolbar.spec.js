import { shallowMount, mount } from '@vue/test-utils';

import { AssessmentItemToolbarActions } from '../../constants';
import AssessmentItemToolbar from './AssessmentItemToolbar';

describe('AssessmentItemToolbar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AssessmentItemToolbar);
  });

  it('smoke test', () => {
    const wrapper = shallowMount(AssessmentItemToolbar);

    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('uses a correct item label in menu if specified', () => {
    wrapper = mount(AssessmentItemToolbar, {
      propsData: {
        itemLabel: 'question',
      },
    });

    expect(wrapper.find('[data-test="toolbarMenu"]').html()).toContain('Add question above');
    expect(wrapper.find('[data-test="toolbarMenu"]').html()).toContain('Add question below');
  });

  it('emits click action with correct payload on arrow up icon click', () => {
    wrapper.find('[data-test=toolbarIconArrowUp]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.MOVE_ITEM_UP);
  });

  it('emits click action with correct payload on arrow down icon click', () => {
    wrapper.find('[data-test=toolbarIconArrowDown]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.MOVE_ITEM_DOWN);
  });

  it('emits click action with correct payload on edit icon click', () => {
    wrapper.find('[data-test=toolbarIconEdit]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.EDIT_ITEM);
  });

  it('emits click action with correct payload on delete icon click', () => {
    wrapper.find('[data-test=toolbarIconDelete]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.DELETE_ITEM);
  });

  it('emits click action with correct payload on "Add item above" menu item click', () => {
    wrapper.find('[data-test=toolbarMenuAddItemAbove]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.ADD_ITEM_ABOVE);
  });

  it('emits click action with correct payload on "Add item below" menu item click', () => {
    wrapper.find('[data-test=toolbarMenuAddItemBelow]').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click.length).toBe(1);
    expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.ADD_ITEM_BELOW);
  });

  it("doesn't render edit icon if displayEditIcon is false", () => {
    wrapper = mount(AssessmentItemToolbar, {
      propsData: {
        displayEditIcon: false,
      },
    });

    expect(wrapper.html()).not.toContain('[data-test="toolbarIconEdit"]');
  });

  describe('when displayDeleteIcon is false', () => {
    beforeEach(() => {
      wrapper = mount(AssessmentItemToolbar, {
        propsData: {
          displayDeleteIcon: false,
        },
      });
    });

    it("doesn't render delete icon", () => {
      expect(wrapper.contains('[data-test="toolbarIconDelete"]')).toBe(false);
    });

    it('renders "Delete" menu item', () => {
      expect(wrapper.contains('[data-test="toolbarMenuDeleteItem"]')).toBe(true);
    });

    it('emits click action with correct payload on "Delete" menu item click', () => {
      wrapper.find('[data-test=toolbarMenuDeleteItem]').trigger('click');

      expect(wrapper.emitted().click).toBeTruthy();
      expect(wrapper.emitted().click.length).toBe(1);
      expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.DELETE_ITEM);
    });
  });

  it("doesn't render menu if displayMenu is false", () => {
    wrapper = mount(AssessmentItemToolbar, {
      propsData: {
        displayMenu: false,
      },
    });

    expect(wrapper.contains('[data-test="toolbarMenu"]')).toBe(false);
  });

  describe('when collapse is true', () => {
    beforeEach(() => {
      wrapper = mount(AssessmentItemToolbar, {
        propsData: {
          collapse: true,
        },
      });
    });

    it("doesn't render arrow up icon", () => {
      expect(wrapper.contains('[data-test="toolbarIconArrowUp"]')).toBe(false);
    });

    it('renders "Move up" menu item', () => {
      expect(wrapper.contains('[data-test="toolbarMenuMoveItemUp"]')).toBe(true);
    });

    it('emits click action with correct payload on "Move up" menu item click', () => {
      wrapper.find('[data-test=toolbarMenuMoveItemUp]').trigger('click');

      expect(wrapper.emitted().click).toBeTruthy();
      expect(wrapper.emitted().click.length).toBe(1);
      expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.MOVE_ITEM_UP);
    });

    it("doesn't render arrow down icon", () => {
      expect(wrapper.contains('[data-test="toolbarIconArrowDown"]')).toBe(false);
    });

    it('renders "Move down" menu item', () => {
      expect(wrapper.contains('[data-test="toolbarMenuMoveItemDown"]')).toBe(true);
    });

    it('emits click action with correct payload on "Move down" menu item click', () => {
      wrapper.find('[data-test=toolbarMenuMoveItemDown]').trigger('click');

      expect(wrapper.emitted().click).toBeTruthy();
      expect(wrapper.emitted().click.length).toBe(1);
      expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.MOVE_ITEM_DOWN);
    });

    it("doesn't render delete icon", () => {
      expect(wrapper.contains('[data-test="toolbarIconDelete"]')).toBe(false);
    });

    it('renders "Delete" menu item', () => {
      expect(wrapper.contains('[data-test="toolbarMenuDeleteItem"]')).toBe(true);
    });

    it('emits click action with correct payload on "Delete" menu item click', () => {
      wrapper.find('[data-test=toolbarMenuDeleteItem]').trigger('click');

      expect(wrapper.emitted().click).toBeTruthy();
      expect(wrapper.emitted().click.length).toBe(1);
      expect(wrapper.emitted().click[0][0]).toEqual(AssessmentItemToolbarActions.DELETE_ITEM);
    });
  });
});
