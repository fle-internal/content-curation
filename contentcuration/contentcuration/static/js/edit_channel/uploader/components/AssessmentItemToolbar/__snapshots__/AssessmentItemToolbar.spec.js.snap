// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentItemToolbar renders 1`] = `
<div style="min-width: 175px;"><button type="button" class="v-btn v-btn--icon theme--light" data-test="toolbarIconEdit">
    <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="color: rgb(104, 104, 104); caret-color: #686868;">edit</i></div>
  </button><span class="v-tooltip v-tooltip--top"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"> <span>Edit</span></div></span> <button type="button" class="v-btn v-btn--icon theme--light" data-test="toolbarIconArrowUp">
  <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="color: rgb(104, 104, 104); caret-color: #686868;">keyboard_arrow_up</i></div>
</button><span class="v-tooltip v-tooltip--top"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"> <span>Move up</span></div></span> <button type="button" class="v-btn v-btn--icon theme--light" data-test="toolbarIconArrowDown">
  <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="color: rgb(104, 104, 104); caret-color: #686868;">keyboard_arrow_down</i></div>
</button><span class="v-tooltip v-tooltip--top"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"> <span>Move down</span></div></span> <button type="button" class="v-btn v-btn--icon theme--light" data-test="toolbarIconDelete">
  <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="color: rgb(104, 104, 104); caret-color: #686868;">close</i></div>
</button><span class="v-tooltip v-tooltip--top"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"> <span>Delete</span></div></span> <button type="button" class="v-btn v-btn--icon theme--light">
  <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light" style="color: rgb(104, 104, 104); caret-color: #686868;">more_vert</i></div>
</button>
<div class="v-menu" data-test="toolbarMenu">
  <div class="v-menu__content theme--light " style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;">
    <div role="list" class="v-list theme--light">
      <!---->
      <!---->
      <div role="listitem"><a data-test="toolbarMenuAddItemAbove" class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__title">Add item above</div>
        </a></div>
      <div role="listitem"><a data-test="toolbarMenuAddItemBelow" class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__title">Add item below</div>
        </a></div>
      <!---->
    </div>
  </div>
</div>
</div>
`;
